source/addonStore/models/addon.py

```diff
@@ -161,13 +161,21 @@ def isPendingInstall(self) -> bool:
 			# have not been installed yet
 			addonDataManager._downloadsPendingInstall,
 		)
+		nameInDownloadsPendingCompletion = filter(
+			lambda m: m.model.name == self.name,
+			# add-ons which are currently being downloaded
+			# and have not been cancelled
+			addonDataManager._downloadsPendingCompletion,
+		)
```

Comment:
this is not safe - an add-on pending install needs to be treated different to an add-on pending completion

Reply:
This is used to exclude download cancellations.

Reply:
yes - but it cannot be implemented as is. We rely on this function in other contexts. We can't treat these things as the same. For example, this tricks NVDA into thinking a downloading add-on is ready for installation. if we attempt to install an add-on that is not pending install it will fail.

Reply:
Then I think the original implementation runs that risk, and we might consider just removing the

```python
			# True if this add-on is currently being downloaded
			# and the download has not been cancelled
			or os.path.exists(self.tempDownloadPath)
```

Reply:
True, can you update the doc string to note that the download might not be completed yet